docs/developer-docs/build/using-an-agent.mdx

```diff
@@ -0,0 +1,56 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import CodeBlock from '@theme/CodeBlock';
+
+# Using an Agent
+
+If you have a canister running, either on your local machine or live on the Internet Computer, you will have two main ways to interact with your canister smart contract. You can talk to the canister using the v2 API using an `agent` that follows the interface specification, or you can use the canister's http interface.
+
+## Calling a Canister
+
+In the Internet Computer ecosystem, a library that is used to make calls to the IC public interface is called an agent. An agent has a few key responsibilities, which make it convenient to work with in your language of choice. 
+
+A list of agents is available at [other languages -> agents](./languages/other-languages/other-languages-intro#interfacing-with-canisters-agents).
+
+### 1. Structuring Data
+
+A `call` to the Internet Computer can take two common forms - an `update` or a `query`. The `agent` submits a POST request to `/api/v2/canister/<effective_canister_id>/call`, and includes the following components: 
+* `request_type`
+* Authentication
+    * `sender`, `nonce`, and `ingress_expiry`
+* `canister_id`
+* `method_name`
+* `arg` - the rest of the payload
+
+By knowing the Candid interface of the canister, the `agent` will assemble the `"arg"` by `cbor`-encoding the data from the client application, and ensuring it matches the Candid interface for the method it will be calling. That encoded data will be turned into a buffer.
+
+The `agent` takes all of that information, bundles it up into a certificate, and attaches it to the body of the POST request. The canister will work on that request asynchronously, and then the `agent` can begin polling with `read_state` requests, until the canister response is ready.
+
+### 2. Decoding Data
+
+Once the data has been returned from the IC, the `agent` takes the certificate from the payload and verifies it. The certificate can be verified as genuine using the public `rootKey` of the NNS subnet. The network will respond with a `cbor`-encoded buffer, which the `agent` can then decode, and transform into a useful structure using semantic language-specific types. For example, if the type returned from the canister is `text`, that will get turned into a JavaScript `string`, and so on.
+
+### 3. Managing Authentication
+
+Calls to the Internet Computer always need to have a cryptographic identity attached. That identity will either be Anonymous or Authenticated, using a cryptographic signature. Since identities are required, canisters can use the identity attached to a call to decide how to respond to that call. This enables contracts to use those identities for other purposes. 
+
+#### Accepted Identities
+
+The IC accepts calls using the following types of identities:
+
+* Ed25519
+* Secp256k1
+* ECDSA with a P-256 curve
```

Comment:
For the self-authenticating principle, should we use the words in the spec?
> ### Ed25519 and ECDSA signatures
> Plain signatures are supported for the schemes
> * [Ed25519](https://ed25519.cr.yp.to/index.html) or
> * [ECDSA](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-4.pdf) on curve P-256 (also known as secp256r1), using SHA-256 as hash function, as well as on the Koblitz curve secp256k1.

Reply:
I modified it slightly, but yes, the precision is better